src/opset/Gemm_13.c

```diff
@@ -0,0 +1,187 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with this program.  If not, see <https://www.gnu.org/licenses/>.
+ */
+
+#include <assert.h>
+#include <math.h>
+#include <stdio.h>
+
+#include <connx/accel.h>
+#include <connx/connx.h>
+
+#define GEMM_DEBUG 0
+
+__attribute__((unused)) static void show_elements_tensor(char* name, int32_t rows, int32_t cols, float32_t* d,
+                                                         bool transposed) {
+    fprintf(stderr, "%s %s Transposed ( %d, %d)\n", name, transposed ? "" : "Not", rows, cols);
+    int32_t num = transposed ? rows : cols;
+    float32_t(*m)[num] = (float32_t(*)[num])d;
+    for (int32_t r = 0; r < rows; r++) {
+        for (int32_t c = 0; c < cols; c++) {
+            float32_t val = transposed ? m[c][r] : m[r][c];
+            fprintf(stderr, "%f ", val);
+        }
+        fprintf(stderr, "\n");
+    }
+    fprintf(stderr, "\n");
+}
+
+// clang-format off
+int Gemm_{{op_version}}(connx_Graph* graph, __attribute__((unused)) uint32_t output_count, uint32_t* outputs,
+                         // clang-format on
+                         __attribute__((unused)) uint32_t input_count, uint32_t* inputs,
+                         __attribute__((unused)) uint32_t attribute_count, __attribute__((unused)) void** attributes) {
+
+    connx_Tensor* A = connx_Graph_get(graph, inputs[0]);
+    connx_Tensor* B = connx_Graph_get(graph, inputs[1]);
+    assert(A != NULL && B != NULL);
+
+    float32_t alpha = *(float32_t*)attributes[0];
+    float32_t beta = *(float32_t*)attributes[1];
+    bool transA = *(int32_t*)attributes[2] ? true : false;
+    bool transB = *(int32_t*)attributes[3] ? true : false;
+
+    float32_t(*a)[A->shape[1]] = A->buffer;
+    float32_t(*b)[B->shape[1]] = B->buffer;
+
+    connx_Tensor* C = NULL;
+    bool is_biased = false;
+
+    if (input_count == 3) {
+        C = connx_Graph_get(graph, inputs[2]);
+        assert(C != NULL);
+        is_biased = true;
+    }
+
+    // alias for A, B rows/cols whether transposed flags enabled
+    int32_t A_rows = A->shape[0];
+    int32_t A_cols = A->shape[1];
+    int32_t B_rows = B->shape[0];
+    int32_t B_cols = B->shape[1];
+    if (transA && transB) {
+        A_rows = A->shape[1];
+        A_cols = A->shape[0];
+        B_rows = B->shape[1];
+        B_cols = B->shape[0];
+    } else if (transA && !transB) {
+        A_rows = A->shape[1];
+        A_cols = A->shape[0];
+    } else if (!transA && transB) {
+        B_rows = B->shape[1];
+        B_cols = B->shape[0];
+    }
+
+    // A's # of cols and B's # of row should be same
+    if (A_cols != B_rows) {
+        connx_error("Gemm: A->shape[1] != B->shape[0]");
+        return CONNX_TENSOR_SHAPE_NOT_MATCHING;
+    }
+
+    // initialize output tensor
+    int ndim = A->ndim; // Always 2D
```

Comment:
If always 2, this will be fine
```suggestion
    int ndim = 2; // Always 2D
```